### DISCUSSION TITLE
scipy.odr

### DISCUSSION THREAD

**Discussion by bytelinker** [lmfit/lmfit-py#789] [General]
Hello,

lmfit is calling scipy.optimize but scipy.odr is not available as solver. Is there an option, to use it? It would be a great advantage to use the wide range of lmfit benefits with this solver, when using uncertainties in both variables.

Regards

bytelinker

**Comment by newville** [comment]
@bytelinker There currently is not an option to use ODR with lmfit.  I agree that it would be an interesting addition.   My recollection is that the interface to ODR is significantly different from `leastsq` and `least_squares`, which is probably why it is in its own odd little box in scipy too. In fact, I think it is oddly "Model"-like and OO.    

Which is to say: it would probably be some work, but if you're willing to give that a try, I would be willing to try to help (though my bandwidth is well past maximum safe operating capacity).

**Reply by bytelinker** [reply]
@newville There are differences as the uncertainty for x has to be an additional parameter. The work might be off better quality when done by more advanced coders.

**Comment by newville** [comment]
@bytelinker Oh, I think that the difference between how one sets up a fit with ODR compared to setting up a fit with `scipy.optimize.minimize` is much bigger than whether there is uncertainty in `x`.   I think adding ODR is sort of not a small task.  I believe it does not handle uncertainties in more than one independent variable.   But, I have not looked at it in a long time.
